packaging/debian-sid/snapd.postrm

```diff
@@ -96,6 +96,10 @@ if [ "$1" = "purge" ]; then
         rm -f "/etc/systemd/system/multi-user.target.wants/$unit"
     done
 
+    # now stop snapd itself to ensure we can clearly remove all snapd data
+    systemctl_stop snapd.socket || true
```

Comment:
also why not stop snapd _before_ we rip out the world underneath it's feet? the logs from when this fails seem to be indicative of the fact that snapd (when run by snap-failure) gets very confused when the world is ripped out from underneath it's feet:
```
Apr 01 18:51:40 apr011848-565010 snap-failure[3170]: snapmgr.go:303: cannot read snap info of snap "snapd" at revision 11402: cannot find installed snap "snapd" at revision 11402: missing file /snap/snapd/11402/meta/snap.yaml
Apr 01 18:51:40 apr011848-565010 snap-failure[3170]: hotplug.go:131: internal error: cannot get global device context: too early for operation, device model not yet acknowledged
... many more of those errors
Apr 01 18:51:41 apr011848-565010 snap-failure[3170]: taskrunner.go:439: DEBUG: Running task 137 on Doing: Consider re-refresh of "snapd", "core18", "lxd", "google-cloud-sdk"
Apr 01 18:51:41 apr011848-565010 snap-failure[3170]: task.go:337: DEBUG: 2021-04-01T18:51:41Z ERROR too early for operation, device not yet seeded or device model not acknowledged
Apr 01 18:51:41 apr011848-565010 snap-failure[3170]: taskrunner.go:271: [change 3 "Consider re-refresh of \"snapd\", \"core18\", \"lxd\", \"google-cloud-sdk\"" task] failed: too early for operation, device not yet seeded or device model not acknowledged
Apr 01 18:51:41 apr011848-565010 snap-failure[3170]: snapmgr.go:303: cannot read snap info of snap "lxd" at revision 19823: cannot find installed snap "lxd" at revision 19823: missing file /snap/lxd/19823/meta/snap.yaml
Apr 01 18:51:41 apr011848-565010 snap-failure[3170]: snapmgr.go:303: cannot read snap info of snap "snapd" at revision 11402: cannot find installed snap "snapd" at revision 11402: missing file /snap/snapd/11402/meta/snap.yaml
Apr 01 18:51:41 apr011848-565010 snap-failure[3170]: snapmgr.go:303: cannot read snap info of snap "core18" at revision 1997: cannot find installed snap "core18" at revision 1997: missing file /snap/core18/1997/meta/snap.yaml
```

Reply:
iirc we did not stop snapd earlier because the stop scripts of snaps (like lxd) might try to poke at snapd via snapctl?